freud/data.py

```diff
@@ -0,0 +1,163 @@
+                dimension. Must be of the form (nx, ny, 1) for 2D boxes
+                (Default value = 1).
+            scale (float): Amount to scale the unit cell by (in distance units)
+                (Default value = 1).
+            sigma_noise (float):
+                The standard deviation of the normal distribution used to add
+                noise to the positions in the system (Default value = 0).
+
+        Returns:
+            tuple (:class:`freud.box.Box`, :class:`np.ndarray`):
+                A system-like object (see
+                :class:`~freud.locality.NeighborQuery.from_system`.).
+        """
+        try:
+            nx, ny, nz = num_replicas
+        except TypeError:
+            nx = ny = num_replicas
+            nz = 1 if self.box.is2D else num_replicas
+
+        if self.box.is2D and nz != 1:
+            raise ValueError("The number of replicas in z must be 1 for a "
+                             "2D unit cell.")
+
+        basis = np.tile(
+            self.basis_positions[np.newaxis, np.newaxis, np.newaxis, ...],
+            (nx, ny, nz, 1, 1))
+        basis[..., 0] += np.arange(nx)[:, np.newaxis, np.newaxis, np.newaxis]
+        basis[..., 1] += np.arange(ny)[np.newaxis, :, np.newaxis, np.newaxis]
+        basis[..., 2] += np.arange(nz)[np.newaxis, np.newaxis, :, np.newaxis]
+
+        positions = (basis - 0.5).reshape(-1, 3).dot(
+            self.box.to_matrix().T)*scale
+        box = freud.box.Box.from_matrix(
+            self.box.to_matrix() * scale * [[nx, ny, nz]])
+
+        if sigma_noise != 0:
+            mean = [0]*3
+            var = sigma_noise*sigma_noise
+            cov = np.diag([var, var, var if self.dimensions == 3 else 0])
+            positions += np.random.multivariate_normal(
+                mean, cov, size=positions.shape[:-1])
+
+        return box, positions
+
+    @property
+    def box(self):
+        """:class:`freud.box.Box`: The box instance containing the lattice
+        vectors."""
+        return self._box
+
+    @property
+    def lattice_vectors(self):
+        """:math:`(3, 3)` :class:`np.ndarray`: The matrix of lattice
+        vectors."""
+        return self._box.to_matrix()
+
+    @property
+    def basis_positions(self):
+        """:math:`(N_{points}, 3)` :class:`np.ndarray`: The basis positions."""
+        return self._basis_positions
+
+    @property
+    def a1(self):
+        """:math:`(3, )` :class:`np.ndarray`: The first lattice vector."""
+        return self.box.to_matrix()[:, 0]
+
+    @property
+    def a2(self):
+        """:math:`(3, )` :class:`np.ndarray`: The second lattice vector."""
+        return self.box.to_matrix()[:, 1]
+
+    @property
+    def a3(self):
+        """:math:`(3, )` :class:`np.ndarray`: The third lattice vector."""
+        return self.box.to_matrix()[:, 2]
+
+    @property
+    def dimensions(self):
+        """int: The dimensionality of the unit cell."""
+        return self.box.dimensions
```

Comment:
Is it better to use `self.box` or `self._box`? In `lattice_vectors`, it uses `self._box`. Be consistent.
```suggestion
        return self.box.dimensions
```